### ISSUE TITLE
Support for out-of-source builds

### ISSUE THREAD

**Issue by lpenz** [eddyxu/cpp-coveralls#61]
The utility assumes that the .c files and the git repository are in the same tree as the .gcno files (-r option).
With cmake, it is quite easy (and common) to build utilities in a separate "build" directory.
I think a --source-root option would make sense.

**Comment by zbeekman** [comment]
huh, so am I correct in saying that if you have:

```
-project_root
|-src
|-other_stuff
|-build
    |-compiler_1
```
you won’t run into any issues?

**Comment by lpenz** [comment]
I'd say that I have to test it, but I think that is correct: it would work if I "-r project_root"

**Comment by 0xced** [comment]
Yes, this is correct, having both `src` and `build` directories under a common parent *root* directory works fine.

**Comment by AndreLouisCaron** [comment]
I'm also trying to use this with CMake on Travis.  The problem I'm getting is that the exclude folders for the coverage reports and the source files is the same.  I keep getting these matches for source files on CMake internal files (`feature_tests.c`, `CompilerId*`).  I can use some specific rules but it seems like this is depending on CMake internals.

Any way we can have exclusion patterns specifically for source file discovery?

**Comment by zbeekman** [comment]
I've found better functionality and support on codecov.io and have moved there. The bash script works quite well, and is easier to use than coveralls.

**Comment by mathstuf** [comment]
Would it be easy to just have two arguments, one for the build tree and another for the source tree?
